views/js/dashboard.js

```diff
@@ -0,0 +1,129 @@
+function init() {
+    var currUser;
+    $.ajax({
+        url: '/getCurrUser',
+        type: 'GET',
+        dataType: 'json',
+        xhrFields: {
+            withCredentials: true
+        },
+        success: function (data) {
+            if (!data)
+                alert("Cannot get current user details. Please try to login again.")
+            else {
+                currUser = data;
+                if (data.img != null)
```

Comment:
what if data is only null then data.img will result in error 
Moreover if data is assigned to currUser then currUser is not used data is used here instead.
Lastly. name, _id are used directly without any check.
So please add a check you have to ensure first if data is there then you have to ensure if fields in the data is there you can do that by simple   
`data && data.hasOwnProperty('img') && so on ....`